src/app/services/count.service.ts

```diff
@@ -116,6 +115,21 @@ export class CountService {
     });
   }
 
+  /**
+   * When no observation counts requested, takes advantage of subject set constraint knowing number of subjects.
+   * Falls back to making reqular call otherwise
+   * @param constraint
+   */
+  getCounts(constraint: Constraint) {
```

Comment:
I am still thinking whether it is better to keep this logic here or move it to transmart-resource.service, but maybe for now it is fine like this and later we will find some better place for all the optimization-related logic.